### PR TITLE
Grid snap

### LINKED ISSUE #368: Snap to grid
I want to be able to snap objects to the grid.

### PR DESCRIPTION
**Pull request by 1uk45**
fixes #368

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L177](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f3c36e8acac3907e56b2418f0523b47f27aa1dff/src/Engine/Hymn.cpp#L177): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L177](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/7bdab4a075d3f1b3de1e566b97fa2b4a9d7dcead/src/Engine/Hymn.cpp#L177): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Editor/GUI/Editors/EntityEditor.cpp [hunk 1]
```diff
@@ -76,6 +77,44 @@ void EntityEditor::Show() {
         ImGui::Text("Transform");
         ImGui::ShowHelpMarker("The entity's position, rotation and scale.", 75.f);
         ImGui::Indent();
+
+        if (Hymn().gridSettings.gridSnap) {
+
```
Comment (andens) [inline src/Editor/GUI/Editors/EntityEditor.cpp]: Not sure about this line break; style doesn't say anything explicitly so I don't know how we want it.

File: src/Engine/Hymn.hpp [hunk 2]
```diff
@@ -107,6 +107,8 @@ class ActiveHymn {
         {
             int gridScale = 0;
             bool gridSettingsOpen = true;
+            bool gridSnap = true;
+            int snapOption = 100;
```
Comment (andens) [inline src/Engine/Hymn.hpp]: Could you remove the empty line at the end and fix opening brace while at it?

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L177](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/339b1f24afdc3d2414e51c597d12f2d73f6ed4f3/src/Engine/Hymn.cpp#L177): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L175](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/5f3f5325be2fe83898d12e0c72e351558761c2f4/src/Engine/Hymn.cpp#L175): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L175](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/4ad85ec443c728b3c258933206a9de113a8b342f/src/Engine/Hymn.cpp#L175): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L178](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2921a072181d869a4228b1f005b692256d9c129c/src/Engine/Hymn.cpp#L178): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L176](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/3f43ff5aacab14e11b3807acac040c1758c576a1/src/Engine/Hymn.cpp#L176): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**